source/common/http/async_client_impl.h

```diff
@@ -195,9 +195,7 @@ class AsyncStreamImpl : public virtual AsyncClient::Stream,
   RequestTrailerMap& addDecodedTrailers() override { PANIC("not implemented"); }
   void addDecodedData(Buffer::Instance&, bool) override {
     // This should only be called if the user has set up buffering. The request is already fully
-    // buffered. Note that this is only called via the async client's internal use of the router
-    // filter which uses this function for buffering.
-    ASSERT(buffered_body_ != nullptr);
```

Comment:
Is this no longer needed?

Reply:
It turns out async lb has some race conditions with streaming non-buffered mirroring which are better documented in the follow-up PR.  that said I don't think this change is needed here - better to land it with the docs.